easybuild/easyblocks/m/mpich.py

```diff
@@ -0,0 +1,104 @@
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_MPICH(ConfigureMake):
+    """
+    Support for building the MPICH MPI library and derivatives.
+    - basically redefinition of environment variables
+    """
+
+    # There is a number of configuration options that are typically needed that are not present
+    # here. The reason is that this easyblock is intended to be used as a parent for other 
+    # easyblocks like MVAPICH2 and PSMPI. Not all of them support the same options, so they
+    # are not included here.
+    def configure_step(self):
+        """Renaming of various environment variables needed for the configuration of MPICH"""
+
+        # things might go wrong if a previous install dir is present, so let's get rid of it
+        if not self.cfg['keeppreviousinstall']:
+            self.log.info("Making sure any old installation is removed before we start the build...")
+            super(EB_MPICH, self).make_dir(self.installdir, True, dontcreateinstalldir=True)
+
+        # MPICH configure script complains when F90 or F90FLAGS are set,
+        # they should be replaced with FC/FCFLAGS instead.
+        # Additionally, there are a set of variables (FCFLAGS among them) that should not be set at configure time,
+        # or they will leak in the mpix wrappers.
+        # Specific variables to be included in the wrapper exists, but they changed between MPICH 3.1.4 and MPICH 3.2
+        # and in a typical scenario we probably don't want them.
+        env_var_tuples = [("F90", "FC"), ("F90FLAGS", "FCFLAGS")]
+        env_var_tuples += [("CFLAGS", "MPICHLIB_CFLAGS"), ("CPPFLAGS", "MPICHLIB_CPPFLAGS")]
+        env_var_tuples += [("CXXFLAGS", "MPICHLIB_CXXFLAGS"), ("FFLAGS", "MPICHLIB_FFLAGS")]
+        env_var_tuples += [("FCFLAGS", "MPICHLIB_FCFLAGS")]
+        env_var_tuples += [("LDFLAGS", "MPICHLIB_LDFLAGS"), ("LIBS", "MPICHLIB_LIBS")]
+        vars_to_unset = []
+        for (envvar, new_envvar) in env_var_tuples:
+            envvar_val = os.getenv(envvar)
+            if envvar_val:
+                new_envvar_val = os.getenv(new_envvar)
+                vars_to_unset += [envvar]
+                if envvar_val == new_envvar_val:
+                    self.log.debug("$%s == $%s, just defined $%s as empty", envvar, new_envvar, envvar)
+                elif new_envvar_val is None:
+                    env.setvar(new_envvar, envvar_val)
+                else:
+                    raise EasyBuildError("Both $%s and $%s set, can I overwrite $%s with $%s (%s) ?",
+                                         envvar, new_envvar, new_envvar, envvar, envvar_val)
+        env.unset_env_vars(vars_to_unset)
+
+        super(EB_MPICH, self).configure_step()
+
+    # make and make install are default
+
+    def sanity_check_step(self, additional_paths=None):
```

Comment:
Mmmh, there are not just dirs, there can be also files. I chose `additional_paths` to make it the same as `custom_paths`.